### DISCUSSION TITLE
sharing of .data and .rego among multiple OPA instances

### DISCUSSION THREAD

**Discussion by vimalagrawal** [open-policy-agent/community#172] [Community]
We would like to create multiple OPA server instances running in parallel on multi cores for better performance. Can these symmetric instances share .rego and .data so that memory footprint is limited. Do they use shared memory?

Vimal

**Comment by anderseknert** [comment]
Hi there! 👋

> We would like to create multiple OPA server instances running in parallel on multi cores for better performance.

That would not result in better performance. OPA will by default use as many cores as is made available to the application, or set by [GOMAXPROCS](https://www.openpolicyagent.org/docs/latest/policy-performance/#resource-utilization).

Multiple OPA instances do not share memory.
